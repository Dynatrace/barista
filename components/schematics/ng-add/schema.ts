export interface Schema {
  /** Whether to install animationsmodule */
  animations: boolean;

  /** Which styles should be installed */
  stylesPack: boolean;

  /** Whether to install dynatrace iconpack */
  iconpack: boolean;

  /** Whether to setup angularJson */
  angularJsonSetup: boolean;
}
