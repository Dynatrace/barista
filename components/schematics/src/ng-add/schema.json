{
  "$schema": "http://json-schema.org/schema",
  "id": "NgAddOptionsSchema",
  "type": "object",
  "description": "Schema for Ng add options",
  "properties": {
    "project": {
      "type": "string",
      "description": "Name of the project where the Barista Components should be set up.",
      "$default": {
        "$source": "projectName"
      }
    },
    "animations": {
      "type": "boolean",
      "default": true,
      "description": "Whether Angular browser animations should be set up.",
      "x-prompt": "Set up browser animations for Barista Components?"
    },
    "typography": {
      "type": "boolean",
      "default": false,
      "description": "Whether to set up global typography styles.",
      "x-prompt": "Set up global Barista typography styles?"
    }
  },
  "required": [],
  "additionalProperties": false
}
