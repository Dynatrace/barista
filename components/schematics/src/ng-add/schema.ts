/**
 * @license
 * Copyright 2019 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export interface Schema {
  /** Whether its a test environment */
  isTestEnv: boolean;

  /** Whether to install animationsmodule */
  animations: boolean;

  /** Which styles should be installed */
  stylesPack: boolean;

  /** Whether to install dynatrace iconpack */
  iconpack: boolean;

  /** Whether to setup angularJson */
  angularPathRefactor: boolean;
}
