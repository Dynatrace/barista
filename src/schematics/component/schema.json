{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsDtComponent",
  "title": "Dt Component Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "spec": {
      "type": "boolean",
      "description": "Specifies if a spec file is generated.",
      "default": true
    },
    "selector": {
      "type": "string",
      "format": "html-selector",
      "description": "The selector to use for the component."
    }
  },
  "required": [
    "name"
  ]
}