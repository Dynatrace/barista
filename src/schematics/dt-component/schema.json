{
  "$schema": "http://json-schema.org/schema",
  "id": "SchemanticsForDtComponent",
  "title": "DtComponent Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "uitest": {
      "type": "boolean",
      "default": false,
      "aliases": ["ui"]
    },
    "universal": {
      "type": "boolean",
      "default": true,
      "aliases": ["un"]
    }
  }
}