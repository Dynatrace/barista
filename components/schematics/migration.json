{
  "$schema": "../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "update-5.0.0": {
      "version": "5.0.0",
      "description": "Updates from the old @dynatrace/angular-components to the new @dynatrace/barista-components",
      "factory": "./src/ng-update/index#updateToV5"
    },
    "ng-post-update": {
      "description": "Prints out results after ng-update.",
      "factory": "./src/ng-update/index#postUpdate",
      "private": true
    }
  }
}
