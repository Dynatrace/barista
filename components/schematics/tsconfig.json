{
  "compilerOptions": {
    "lib": ["es2017"],
    "module": "commonjs",
    "moduleResolution": "node",
    "noEmitOnError": false,
    "strictNullChecks": true,
    "skipDefaultLibCheck": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "skipLibCheck": true,
    "sourceMap": true,
    "declaration": true,
    "target": "es6",
    "types": ["jest", "node"],
    "resolveJsonModule": true,
    "baseUrl": ".",
    "paths": {
      "@angular/cdk/schematics": ["node_modules/@angular/cdk/schematics"]
    }
  },
  "include": ["**/*.ts", "migration.json"],
  "exclude": [
    "**/*.spec.ts",
    // Exclude all test-case files because those should not be included in the schematics output.
    "ng-update/tests/**/*.ts"
  ]
}
