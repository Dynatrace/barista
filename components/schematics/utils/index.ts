export * from './testing';
export * from './fileSearch';
