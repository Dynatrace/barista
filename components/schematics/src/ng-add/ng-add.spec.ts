/**
 * @license
 * Copyright 2019 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Tree } from '@angular-devkit/schematics';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { Schema } from './schema';
import {
  createPackageTree,
  createMultipleImportsTree,
  createAngularJsonTree,
  createPeerDependenciesTree,
  createAppModuleTree,
  createStyleCssTree,
  styleCss,
  ANGULARJSON,
  APPMODULE,
  PACKAGEJSON,
  PEERDEPENDENCIES,
  RENAMEDIMPORT,
} from './testing-constants';

const collectionPath = join(__dirname, '../collection.json');

const testRunner = new SchematicTestRunner('schematics', collectionPath);

const schematicOptions: Schema = {
  isTestEnv: true,
  angularPathRefactor: true,
  animations: true,
  iconpack: true,
  stylesPack: true,
};

/**
 * Schematic helper function for starting ng-add schematic
 * @param schematicName Name of the schematic
 * @param options Options concerning the ng-add schematic
 * @param tree host tree
 */
export async function runSchematic(
  schematicName: string,
  options: any,
  t: Tree,
): Promise<UnitTestTree> {
  return testRunner.runSchematicAsync(schematicName, options, t).toPromise();
}

/**
 * Retrieve content of certain path in host tree
 * @param tree The tree from host
 * @param path The path to file
 */
export function getFileContent(t: Tree, path: string): string {
  const fileEntry = t.get(path);

  if (!fileEntry) {
    throw new Error(`The file (${path}) does not exist.`);
  }

  return fileEntry.content.toString();
}

let tree: Tree;
// Testing of Dynatrace Ng-Add Schematic
describe('ng-add schematic for dynatrace barista-components', () => {
  describe('Updating imports from @dynatrace/angular-components to @dynatrace/barista-components.', () => {
    beforeAll(async () => {
      tree = Tree.empty();
      createPackageTree(tree);
      createMultipleImportsTree(tree);
      createAngularJsonTree(tree);
      createPeerDependenciesTree(tree);
      createAppModuleTree(tree);
      createStyleCssTree(tree);

      await runSchematic('ng-add', schematicOptions, tree);
    });

    it('should update imports in package.json', () => {
      expect(getFileContent(tree, '/package.json')).toEqual(PACKAGEJSON);
    });

    it('should update peerDependecies', () => {
      expect(getFileContent(tree, 'components/src/peerPackage.json')).toEqual(
        PEERDEPENDENCIES,
      );
    });

    it('should update imports in project', () => {
      expect(getFileContent(tree, 'apps/src/main.ts')).toEqual(RENAMEDIMPORT);
    });

    it('should install @angular/animations', () => {
      expect(getFileContent(tree, 'app.module.ts')).toEqual(APPMODULE);
    });

    it('should insert styles to main css file', () => {
      expect(getFileContent(tree, 'index.css')).toEqual(styleCss);
    });

    it('should update icons and fonts paths in angular.json', () => {
      expect(getFileContent(tree, 'angular.json')).toEqual(ANGULARJSON);
    });
  });
});
