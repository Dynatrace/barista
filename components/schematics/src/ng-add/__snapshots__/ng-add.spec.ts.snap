// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ng-add schematic for dynatrace barista-components should add @angular/platform-browser-dynamic 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "8.2.3",
    "@angular/common": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@angular/platform-browser-dynamic": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
}
`;

exports[`ng-add schematic for dynatrace barista-components should add all the necessary peer dependencies if no barista or angular components are installed 1`] = `
Object {
  "dependencies": Object {
    "@angular/cdk": "^8.2.3",
    "@angular/common": "^8.2.12",
    "@angular/compiler": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
  "devDependencies": Object {
    "@angular-devkit/build-angular": "^0.803.15",
    "@angular-devkit/build-webpack": "^0.803.15",
    "@angular-devkit/schematics": "^8.3.15",
    "@angular/cli": "^8.3.15",
  },
}
`;

exports[`ng-add schematic for dynatrace barista-components should add barista icons to the angular.json 1`] = `
Object {
  "cli": Object {
    "defaultCollection": "@nrwl/angular",
  },
  "defaultProject": "myapp",
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "options": Object {
            "aot": false,
            "assets": Array [
              "apps/myapp/src/favicon.ico",
              "apps/myapp/src/assets",
              Object {
                "glob": "**/*",
                "input": "node_modules/@dynatrace/barista-fonts/fonts/",
                "output": "/fonts",
              },
            ],
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/apps/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "scripts": Array [],
            "styles": Array [
              "node_modules/@dynatrace/barista-components/style/index.scss",
              "node_modules/@dynatrace/barista-fonts/typography.scss",
              "apps/myapp/src/styles.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp",
      "sourceRoot": "apps/myapp/src",
    },
    "myapp-e2e": Object {
      "architect": Object {
        "e2e": Object {
          "builder": "@nrwl/cypress:cypress",
          "configurations": Object {
            "production": Object {
              "devServerTarget": "myapp:serve:production",
            },
          },
          "options": Object {
            "cypressConfig": "apps/myapp-e2e/cypress.json",
            "devServerTarget": "myapp:serve",
            "tsConfig": "apps/myapp-e2e/tsconfig.e2e.json",
          },
        },
        "lint": Object {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": Object {
            "exclude": Array [
              "**/node_modules/**",
              "!apps/myapp-e2e/**",
            ],
            "tsConfig": Array [
              "apps/myapp-e2e/tsconfig.e2e.json",
            ],
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp-e2e",
      "sourceRoot": "apps/myapp-e2e/src",
    },
    "some-lib": Object {
      "architect": Object {
        "lint": Object {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": Object {
            "exclude": Array [
              "**/node_modules/**",
              "!libs/some-lib/**",
            ],
            "tsConfig": Array [
              "libs/some-lib/tsconfig.lib.json",
              "libs/some-lib/tsconfig.spec.json",
            ],
          },
        },
        "test": Object {
          "builder": "@nrwl/jest:jest",
          "options": Object {
            "jestConfig": "libs/some-lib/jest.config.js",
            "setupFile": "libs/some-lib/src/test-setup.ts",
            "tsConfig": "libs/some-lib/tsconfig.spec.json",
          },
        },
      },
      "prefix": "mc",
      "projectType": "library",
      "root": "libs/some-lib",
      "schematics": Object {},
      "sourceRoot": "libs/some-lib/src",
    },
  },
  "version": 1,
}
`;

exports[`ng-add schematic for dynatrace barista-components should add styles correctly even if there is no styles array in the angular.json  1`] = `
Object {
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "options": Object {
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/apps/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "styles": Array [
              "node_modules/@dynatrace/barista-components/style/index.scss",
              "node_modules/@dynatrace/barista-fonts/typography.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp",
      "sourceRoot": "apps/myapp/src",
    },
  },
  "version": 1,
}
`;

exports[`ng-add schematic for dynatrace barista-components should add the \`@angular/animations\` package with the same version as the \`@angular/core\` package when specified 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "^8.2.3",
    "@angular/common": "^8.2.12",
    "@angular/compiler": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@angular/platform-browser-dynamic": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
  "devDependencies": Object {
    "@angular-devkit/build-angular": "^0.803.15",
    "@angular-devkit/build-webpack": "^0.803.15",
    "@angular-devkit/schematics": "^8.3.15",
    "@angular/cli": "^8.3.15",
  },
}
`;

exports[`ng-add schematic for dynatrace barista-components should not add styles and assets if he has a legacy version of the angular-components installed 1`] = `
Object {
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "options": Object {
            "aot": false,
            "assets": Array [
              "apps/myapp/src/favicon.ico",
              "apps/myapp/src/assets",
              Object {
                "glob": "**",
                "input": "node_modules/@dynatrace/angular-components/assets",
                "output": "/cache",
              },
              Object {
                "glob": "*.svg",
                "input": "node_modules/@dynatrace/dt-iconpack",
                "output": "/cache/icons",
              },
            ],
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/apps/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "scripts": Array [],
            "styles": Array [
              "apps/myapp/src/styles.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp",
      "sourceRoot": "apps/myapp/src",
    },
  },
  "version": 1,
}
`;

exports[`ng-add schematic for dynatrace barista-components should update imports of @dynatrace/angular-components to barista-components in package.json 1`] = `
Object {
  "dependencies": Object {
    "@angular/cdk": "8.2.3",
    "@angular/common": "8.2.3",
    "@angular/core": "8.2.3",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
}
`;

exports[`ng-add schematic for dynatrace barista-components shouldn't add @angular/animations\` package if already installed 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "8.2.3",
    "@angular/common": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@angular/platform-browser-dynamic": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
}
`;
